docs/migration_to_content_publisher/locked-documents.md

```diff
@@ -0,0 +1,52 @@
+
+### Database
+
+While unlikely, we decided to protect against the possibility that a user could attempt to modify an edition at the database level (see [PR](https://github.com/alphagov/whitehall/pull/4936)).  We introduced a `LockedDocumentConcern` which provides a check if a document is locked and raises an exception if it is.  It was obvious this check should apply to the `Edition` and `Document` models, but since it has a large number of associations we needed to consider whether the check should apply to those models as well.  We determined that it would be excessive to introduce this check for every possible model, particularly since we think there is a very low risk of this issues occurring.  However we identified that it would be very problematic if a locked document was featured or added to a document collection, so we've included the check in the `Feature` and `DocumentCollectionGroupMembership` models too.
+
+### External API calls
+
+In the unlikely event that a locked document is edited despite the aforementioned precautions, we need to ensure that any changes that are made are not synced to the various external APIs that Whitehall communicates with.  The affected APIs we identified are Publishing API, Search API and Asset Manager.  We have introduced checks to prevent locked document data from being sent to these APIs.  In some cases, such as Publishing API, Whitehall makes calls from many different places in the application so we have made a best effort to identify the key areas.  There may be some risk that we have missed out a few places but we expect it to be unlikely that locked document data would make it to these APIs given the checks that have been implemented at the model, view and controller levels.  A more detailed account of the approach taken can be found in [this PR](https://github.com/alphagov/whitehall/pull/4930).
+
+### Featured documents
+
+Users can choose to feature a document on an organisation, world location or topical event page.  Currently only documents from Whitehall's own database can be featured and once a document has been migrated to Content Publisher then the link to this document will become out of sync and may break.  The initial plan is to replace existing featured links to migrated documents with an external link.  However we want to prevent locked documents from being able to be featured at all so that we don't have to periodically replace links.  In [this PR](https://github.com/alphagov/whitehall/pull/4920) we adapted the `Admin::EditionFilter` to exclude locked documents from the filter results by default so that they don't appear in the list of documents to be featured.  We also ensured that at the controller level an exception is raised if a `Feature` is created with a locked document.
```

Comment:
I think we already flag the exception earlier in this document so I don't think you need the last sentence, rest sounds good though 👍